src/large_scale_condensation.jl

```diff
@@ -1,46 +1,99 @@
+                                    model::PrimitiveEquationModel,
+                                    ) where {NF<:AbstractFloat}
+
+    get_saturation_vapour_pressure!(column, model)
+    get_saturation_specific_humidity!(column, model)
+    
+    @unpack gravity, RH_thresh_max, RH_thresh_range, RH_thresh_boundary, humid_relax_time = model.constants
+    @unpack cp, alhc, k_lsc = model.parameters
+    @unpack σ_levels_full, σ_levels_thick = model.geometry
+
+    @unpack humid, pres = column            # prognostic variables: specific humidity, surface pressure
+    @unpack temp_tend, humid_tend = column  # tendencies to write into
 
     # 1. Tendencies of humidity and temperature due to large-scale condensation
-    for k = k_lsc:nlev  # Used to be 2:nlev in original Speedy
-        σₖ = σ_levels_full[k]
-        RH_threshold = RH_thresh_max + RH_thresh_range * (σₖ^2 - 1)  # Relative humidity threshold for condensation (Formula 24)
+    # skip the planetary boundary layer, k_lsc is lowest level for precip 
+    for k in eachlayer(column)[k_lsc:end]
+        
+        # Relative humidity threshold for condensation (Formula 24)
+        σₖ² = σ_levels_full[k]^2
+        RH_threshold = RH_thresh_max + RH_thresh_range * (σₖ² - 1)
         if k == nlev
             RH_threshold = max(RH_threshold, RH_thresh_boundary)
         end
 
         # Impose a maximum heating rate to avoid grid-point storm instability
         # This formula does not appear in the original Speedy documentation
-        humid_tend_max = 10σₖ^2 / 3600humid_relax_time
-
-        for j = 1:nlat, i = 1:nlon
-            humid_threshold = RH_threshold * sat_spec_humidity[i, j, k]  # Specific humidity threshold for condensation
-            if humid_grid[i, j, k] > humid_threshold
-                humid_tend_lsc[i, j, k] = -(humid_grid[i, j, k] - humid_threshold) / humid_relax_time                 # Formula 22
-                temp_tend_lsc[i, j, k] = -alhc / cp * min(humid_tend_lsc[i, j, k], humid_tend_max * pres_grid[i, j])  # Formula 23
-                cloud_top[i, j] = min(cloud_top[i, j], k)                                                             # Page 7 (last sentence)
-            else
-                humid_tend_lsc[i, j, k] = zero(NF)
-                temp_tend_lsc[i, j, k] = zero(NF)
-            end
+        humid_tend_max = 10σₖ² / 3600humid_relax_time
+        humid_threshold = RH_threshold * sat_spec_humidity[k]  # Specific humidity threshold for condensation
+
+        if humid[k] > humid_threshold
+            # accumulate in tendencies (nothing is added if humidity not above threshold)
+            humid_tend[k] += -(humid[k] - humid_threshold) / humid_relax_time       # Formula 22
+            temp_tend[k] += -alhc / cp * min(humid_tend[k], humid_tend_max*pres)    # Formula 23
+            column.cloud_top = min(column.cloud_top, k)                             # Page 7 (last sentence)
         end
```

Comment:
The `else` case is not needed anymore as for every grid point we start with zeros in `humid_tend, temp_tend` and can therefore just accumulate in. In `ColumnVariables` `cloud_top` is now just an `Int=0`, so if I understand line 35 correctly, we should initialize this variable with `nlev+1` and only if the threshold is exceed for a lower layer then `cloud_top` is changed? Note that scalars like `column.cloud_top` are immutable (although `column` is mutable) hence the typical `@unpack cloud_top = column` will not update `column.cloud_top`, hence the explicit `column.cloud_top` here.

Reply:
resolved with later commits.

Reply:
> In `ColumnVariables` `cloud_top` is now just an `Int=0`, so if I understand line 35 correctly, we should initialize this variable with `nlev+1` and only if the threshold is exceed for a lower layer then `cloud_top` is changed? 

This is from fortran speedy where, if I remember correctly, `cloud_top == nlev + 1` means no clouds. We could easily change that.

> Note that scalars like `column.cloud_top` are immutable (although `column` is mutable) hence the typical `@unpack cloud_top = column` will not update `column.cloud_top`, hence the explicit `column.cloud_top` here.

Good to know, that could easily lead to some nasty bugs!

Reply:
Yes we now have cloud_top = nlev+1 at initialisation of `ColumnVariables`, otherwise the `min(cloud_top,k)` doesn't work.